tests/test_csv2pq.py

```diff
@@ -0,0 +1,183 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+'''
+This program excersises the csv2pq command to make sure it works as
+expected.
+'''
+
+import os
+import subprocess
+import unittest
+
+
+# *****************************************************************************
+# *                              T e s t I n f o                              *
+# *****************************************************************************
+
+class TestInfo:
+
+    hush = True   # Set to False to see all test output in real time
+
+    inFiles = ["data/testfile1.csv", "data/testfile2.csv"]
+    inSchem = "data/test.schema"
+    outFile = ["/tmp/testfile1.pq", "/tmp/testfile2.pq"]
+    numTest = 1
+
+
+# *****************************************************************************
+# *                              c h k 4 F i l e                              *
+# *****************************************************************************
+
+def chk4File(flist, rm=False):
+    "Verify that created output files actually exist."
+
+    for file in flist:
+        if not os.path.exists(file):
+            print('Output file "' + file + '" not found.')
+            return False
+        if (rm): os.unlink(file)
+    return True
+
+
+# *****************************************************************************
+# *                               r u n T e s t                               *
+# *****************************************************************************
+
+def runTest(cmd, indata=None):
+    "Run a command to test it."
+
+    nTest = str(TestInfo.numTest)
+    TestInfo.numTest += 1
+    print("Running test", nTest + ':', cmd)
+
+    # We would use the newfangled subprocess options to do this but those
+    # are broken in various versions of python 3 so are undependable.
+    #
+    if indata:
+        rPipe, wPipe = os.pipe()
+        os.write(wPipe, bytes(indata, 'ascii'))
+        os.close(wPipe)
+    else:
+        rPipe = None
+
+    try:
+        status = subprocess.run(cmd.split(), stdin=rPipe)
+    except Exception as err:
+        print('Test', nTest, 'ended with an exception;', err)
+        return None
+
+    print("Test", nTest, 'ended with RC =', status.returncode)
+    print('-' * 79)
+    return status.returncode == 0
+
+
+# *****************************************************************************
+# *                          s e t u p _ m o d u l e                          *
+# *****************************************************************************
+
+def setup_module():
```

Comment:
Well, I cheat because I was told that pytest was good enough. Is that wrong? Apparently, not according to what you said.

Reply:
We use pytest to run tests (and `setup_module` is acceptable) but the DM standard is to write tests using `unittest`. Some SQuaRE packages have requested an exception to use pure pytest and that has been granted.

Reply:
@timj, does it mean that we cant use pure unittest (like in `python -m unittest tests/test_whatever.py`)? That would ignore `setup_module` which I think is supposed to do something important for all tests? I do prefer sometimes to run tests without pytest, pytest output may get very unwieldy.

Reply:
In science pipelines we set it up so that `python tests/test_whatever.py` works because there is a `__main__` check that calls `unittest.main()` so it's even easier than using `-m`. The `setup_module` is used if you require pytest to initialize things (which is needed for the memory checker and file leak checker).